src/service/group-service.js

```diff
@@ -0,0 +1,175 @@
+
+        const allGroups= groupRepository.GetAllGroup(skip,take,orderBy,groupname);
+        return allGroups;
+    } 
+
+    //특정 그룹 가져오기
+    getGroupById = async(Id) => {
+        const group = groupRepository.GetGroupById(Id)
+        return group;
+    }
+
+    // 닉네임과 비밀번호 검증, tag와 group, user 수정(트랜잭션 구현 필요)
+    modifyGroup = async (data) => {
+        const {groupId,name, description,
+                ownerNickname, ownerPassword, 
+                photoUrl, tags, goalRep, 
+                discordWebhookUrl, discordInviteUrl} = data
+
+           
+                // const data = {groupId,name, description,
+                // ownerNickname, ownerPassword, 
+                // photoUrl, tags, goalRep, 
+                // discordWebhookUrl, discordInviteUrl}
+
+        const groupPassword = await groupRepository.GetPassword(groupId);
+        const groupNickname = await groupRepository.GetNickname(groupId);
+        
+        const prismaData = {
+                    group_name: name,
+                    description,
+                    goal_rep: goalRep,
+                    discord_webhook_url: discordWebhookUrl,
+                    discord_invite_url : discordInviteUrl
+                }
+                
+
+        //이미지 구현과 연동 필요
+        if (groupNickname != ownerNickname) {
+            let error = new Error;
+            error.statusCode = 401;
+            error.message = "wrong nickname"
+            error.path = 'nickname'
+            throw error;
+        }else{
+            if (groupPassword == ownerPassword ){
+            const modifiedGroup = await groupRepository.PatchGroup(prismaData, groupId);
+
+            let newTags;
+            if (tags){   
+                const deleteTagIds = await tagRepository.deleteTagsbyGroupId(groupId);
+
+                newTags = await tagRepository.createTagsbyTagNames(tags,groupId);
+            
+                const result = [modifiedGroup,newTags];
+                return result;
+
+            }else{
+                let error = new Error;
+                error.statusCode = 401;
+                error.message = "wrong password"
+                error.path = 'password'
+                next(error);
+            }
+        }
+        }
+        
+    }
+
+    // 비밀번호 검증, 그룹, 유저 삭제
+    deleteGroup = async (groupId, inputPassword) => {
+        //삭제할 그룹 찾기 
+        const group = await groupRepository.GetGroupByIdAll(groupId);
+
+        const groupPassword = await groupRepository.GetPassword(groupId);
+        const reqPassword = inputPassword.ownerPassword;
+        //에러처리하기
+
+        if (groupPassword == reqPassword){
+            await groupRepository.DeleteGroup(groupId);
+            console.log("비밀번호 인증 성공")
+            return groupPassword
+        }else{
+            let error = new Error;
+            error.statusCode = 401;
+            error.message = "wrong password"
+            error.path = 'password'
+            throw error;
+        }   
+    }
```

Comment:
```suggestion
    deleteGroup = async (groupId, inputPassword) => {
        const owner = await groupRepository.GetPassword(groupId); // 개선된 repository 메소드 사용
    
        if (!owner) {
            const error = new Error;
            error.statusCode = 403;
            error.message = "Group or Owner not found";
            error.path = 'Group or Owner';
            throw error ;
        }

        if (owner.password !== inputPassword) {
            const error = new Error;
            error.statusCode = 401;
            error.message = "wrong password";
            error.path = 'password';
            throw error ;
        }

        await groupRepository.DeleteGroup(groupId);
    }
```
컨트롤러에서 넘겨서 서비스에서 비즈니스 로직을 처리하는 것이 좋을 것 같습니다!